Data/ByteArray/Sized.hs

```diff
@@ -179,26 +213,6 @@ inlineUnsafeCreate f = unsafeDoIO (alloc @n f)
 empty :: forall ba . ByteArrayN 0 ba => ba
 empty = unsafeDoIO (alloc @0 $ \_ -> return ())
 
--- | Pack a list of bytes into a bytearray
-pack :: forall n ba . (ByteArrayN n ba, KnownNat n) => ListN n Word8 -> ba
-pack l = inlineUnsafeCreate @n (fill $ unListN l)
-  where fill []     _  = return ()
-        fill (x:xs) !p = poke p x >> fill xs (p `plusPtr` 1)
-        {-# INLINE fill #-}
-{-# NOINLINE pack #-}
-
--- | Un-pack a bytearray into a list of bytes
-unpack :: forall n ba
-        . (ByteArrayN n ba, KnownNat n, NatWithinBound Int n, ByteArrayAccess ba)
-       => ba -> ListN n Word8
-unpack bs =  fromMaybe (error "the impossible appened") $ toListN @n $ loop 0
-  where !len = length bs
-        loop i
-            | i == len  = []
-            | otherwise =
-                let !v = unsafeDoIO $ withByteArray bs (`peekByteOff` i)
-                 in v : loop (i+1)
```

Comment:
I moved this code elsewhere so that I didn't repeat the CPP condition all over the place.